File: minesweeper/java-minesweeper/src/main/java/it/chicio/minesweeper/field/resolver/FieldsResolverByIteratingThroughThem.java
package it.chicio.minesweeper.field.resolver;

import it.chicio.minesweeper.field.Field;

import java.util.ArrayList;
import java.util.List;

public class FieldsResolverByIteratingThroughThem implements FieldsResolver {
    private FieldResolver fieldResolver;

    public FieldsResolverByIteratingThroughThem(FieldResolver fieldResolver) {
        this.fieldResolver = fieldResolver;
    }

    public List<Field> resolve(List<Field> fields) {
        ArrayList<Field> resolvedFields = new ArrayList<Field>();
        for (Field field : fields) {
            resolvedFields.add(fieldResolver.resolve(field));
        }
        return resolvedFields;
    }
}

File: minesweeper/kotlin-mockk-minesweeper/src/main/java/it/chicio/minesweeper/field/resolver/FieldsResolverByIteratingThroughThem.kt
package it.chicio.minesweeper.field.resolver

import it.chicio.minesweeper.field.Field
import java.util.*

class FieldsResolverByIteratingThroughThem(private val fieldResolver: FieldResolver) : FieldsResolver {
    override fun resolve(fields: List<Field>): List<Field?> {
        val resolvedFields = ArrayList<Field?>()
        for (field in fields) {
            resolvedFields.add(fieldResolver.resolve(field))
        }
        return resolvedFields
    }
}